tools/metadata_loader/main.py

```diff
@@ -0,0 +1,171 @@
+    data = {
+        "name": f'{args.name}',
+        "accountName": f'{args.accountName}',
+        "containerName": f'{args.containerName}',
+        "description": f'{args.description}'
+    }
+    resp = call_create_datasource_api(url, payload=data)
+
+    print(resp.text)
+
+    return resp.text
+
+
+def call_get_all_metadata_api(url:str):
+    return requests.get(url)
+    
+def get_all_meta(args):
+    
+    config = get_config()
+    
+    url = f'{config["API_URL_BASE"]}/api/datasources/{args.accountName}/{args.containerName}/meta'
+    resp = call_get_all_metadata_api(url)
+
+    items = json.loads(resp.text)
+    if len(items) == 0:
+        print(f"There are no items in account: {args.accountName}, container: {args.containerName}.")
+    else:
+        for item in items:
+            print(f"Account: {item['accountName']}, Container: {item['containerName']}, filepath: {item['filepath'].replace('(slash)', '/')}")
+
+    return resp.text
+
+def call_create_meta_api(url, payload):
+    return requests.post(url, json=payload)
+
+def create_meta(accountName: str, containerName: str, filepath: str, document: str):
+
+    config = get_config()
+
+    #quoted_filepath = quote(filepath, safe='')
+    quoted_filepath = filepath.replace("/", "(slash)")
+    url = f'{config["API_URL_BASE"]}/api/datasources/{accountName}/{containerName}/{quoted_filepath}/meta'
+    resp = call_create_meta_api(url, payload=document)
+    return resp.text
+
+
+def initial_load_meta(args):
+
+    config = get_config()
+
+    storage_url = config["STORAGE_ACCOUNT_URL"]
+    storage_sas = config["STORAGE_SAS_KEY"]
+    blob_service_client = BlobServiceClient(account_url=storage_url, credential=storage_sas)
+    container_client = blob_service_client.get_container_client(container=args.containerName)
+
+    blob_list = container_client.list_blobs()
+
+    overall_response = True
+    for blob in blob_list:
+        
+        # print(blob.name)
+        # if blob.name == "/dir1/dir2/abc.sigmf-meta"
+        # then basename = abc.sigmf-meta, dirname = /dir1/dir2
+
+        basename = os.path.basename(blob.name)
+        parts = basename.split('.')
+        if len(parts)<2 or parts[1] != 'sigmf-meta':
+            continue
+
+        blob_client = container_client.get_blob_client(blob=blob.name)
+        downloader = blob_client.download_blob(max_concurrency=1, encoding='UTF-8')
+        blob_text = downloader.readall()
+
+        dirname = os.path.dirname(blob.name)
+        filepath = f"{dirname}/{parts[0]}"
+
+        resp = create_meta(args.accountName, args.containerName, filepath, blob_text)
+
+        overall_response = overall_response and resp == "Success"
+        
+    return overall_response
+
+
+def start():
+# commands
+# python main.py ...
+#    datasource add -name -accountName -containerName -description
+#    datasource list
+#    metadata list
+#    metadata addfolder -accountName -containerName -document
```

Comment:
There is a convention in python for documentation strings called doc strings that would auytomatically add this to the method __doc__ that can be usefull for things like this.
https://peps.python.org/pep-0257/

Reply:
Thanks for the reminder. done.